ide-support/revhtml5urllibrary.livecodescript

```diff
@@ -0,0 +1,455 @@
+            end if
+            __ulExtSetURLStatus pURL, "error", tError, true
+
+            -- even though the request failed, still store any response from the server
+            ulExtSetLastRHHeaders __ulExtXMLHTTPGetResponseHeaders(pRequest)
+            ulExtSetLastHTTPHeaders __ulExtXMLHTTPGetRequestHeaders(pRequest)
+            ulExtSetAsLoaded pUrl, __ulExtXMLHTTPGetResponse(pRequest)
+
+            ulSendMessage pURL
+            __ulExtCleanupURL pURL
+         end if
+         break
+
+      case "load"
+         if ulExtIsLoadReq(pUrl) then
+            ulExtSetErrorStatus pUrl, empty
+            ulExtSetLastRHHeaders __ulExtXMLHTTPGetResponseHeaders(pRequest)
+            ulExtSetLastHTTPHeaders __ulExtXMLHTTPGetRequestHeaders(pRequest)
+
+            if ulExtIsFileTransfer(pUrl) is not empty then
+               -- XMLHTTP doesn't support download to file, so just write the data to file directy
+               ulExtSetLoadStatus pUrl, "downloaded"
+               put __ulExtXMLHTTPGetResponse(pRequest)  into URL ("binfile:" & ulExtIsFileTransfer(pUrl))
+            else
+               ulExtSetLoadStatus pUrl, "cached"
+               ulExtSetAsLoaded pUrl, __ulExtXMLHTTPGetResponse(pRequest)
+            end if
+
+            ulSendMessage pURL
+            __ulExtCleanupURL pURL
+         end if
+         break
+   end switch
+end ulExtXMLHTTPCallback
+
+--------------------------------------------------------------------------------
+
+private command __ulExtSetURLStatus pURL, pStatus, pMessage, pSendCallback
+   if pStatus is "error" then
+      ulExtSetErrorStatus pUrl, pMessage
+   else
+      ulExtSetErrorStatus pUrl, empty
+   end if
+
+   if ulExtIsLoadReq(pUrl) then
+      ulExtSetLoadStatus pUrl, pStatus
+   end if
+
+   if pSendCallback then
+      if pMessage is not empty then
+         ulSendCallback pURL, pMessage
+      else
+         ulSendCallback pURL, pStatus
+      end if
+   end if
+end __ulExtSetURLStatus
+
+private command __ulExtCleanupURL pURL
+   local tRequest
+   put __ulExtXMLHTTPFromURL(pURL) into tRequest
+   if tRequest is not empty then
+      __ulExtXMLHTTPDestroy tRequest
+   end if
+   ulExtRemoveLoadingVars pURL
+end __ulExtCleanupURL
+
+-- Utililty handler that calls the given function in the javascript url library
+-- Pass the function name and any parameters
+private function __ulExtCallJS pFunction
+   local FUNCTIONNAME
+   put pFunction into FUNCTIONNAME
+
+   -- build the list or parameters escaping any quotes
+   local PARAMLIST
+   repeat with tParamNo = 2 to the paramCount
+      local tParam
+      put param(tParamNo) into tParam
+      if tParam is not a number and tParam is not among the words of "true false" then
+         replace quote with backslash & quote in tParam
+         replace return with "\n" in tParam
+         replace cr with "\r" in tParam
+         replace backslash with "\\" in tParam
```

Comment:
I commented on the commit but it doesn't appear in the PR:
You need to replace backslashes first, otherwise you'll be escaping them in the escape sequences - e.g. an escaped return will end up as '\\\\n' rather than just '\\n'